### DISCUSSION TITLE
How to use dewarp?

### DISCUSSION THREAD

**Discussion by mrobe** [ScanTailor-Advanced/scantailor-advanced#55] [Q&A]
On macOS, I'm having trouble making dewarp work in scan-tailor advanced.

Really basic stuff like what to click, the workflow, etc. is totally unobvious to me. The documentation isn't helpful.

For reference, I'm trying to fix pages that are warped like this: https://freeimage.host/i/HrWMTmP

I've drawn in some red lines so that the warping of the original scan is clear.

Is there a way to set up scan tailor to fix this?

By fiddling around, I can get a grid, but when I drag a corner, nothing happens.

In general, the app feels very slow, even on a dual-core i7. Granted, my input image files are very large, but I don't see any way to reduce their size before the output stage.

Any suggestions?

**Comment by darnn** [comment]
That just looks like that's the way it was printed (note for instance that the lines of text are straight). If you did dewarp it, you would actually be warping it, which doesn't seem great in this specific example. So I'd just leave it as it is. Compare also with the Internet Archive's scan: https://archive.org/details/cinemaasgraphica00nils/page/32/mode/2up

At any rate, this is in Spanish, and an older version, but it should give you an idea: https://youtu.be/pEocwJm7Hj0?t=84

**Comment by mrobe** [comment]
Thanks for your message. I'll have a look at the video, thanks. :)

Sorry, perhaps that wasn't the best example. The diagram is supposed to be a film frame, so the top edge is definitely not warped like that on the printed page. It's just not an artifact of printing.

Perhaps this is a better example from the same book: https://freeimage.host/i/tempimagejtemkh.HrXLUaj

Again, red horizontal lines to show that the upper quarter of the page is warped, while the lower part of the page isn't.

Another example, different book, this time worse, with both upper and lower parts of the page warped: https://freeimage.host/i/HrXQpfe

These are scanning artifacts and, if possible, I'd like to get rid of them.

Question is: can the dewarp function do this job? Automatic setting, or... ?

thanks!

**Comment by darnn** [comment]
Well, going off of my own experience (so make of it what you will): With the first image, I doubt dewarping it would greatly improve it. I'm pretty sure in that situation it's still the printing rather than the scan, it was typeset in the 1950s, when things weren't always perfectly straight, just good enough for the human eye on a printed page. If you've scanned this yourself and you can also try scanning a modern book using the same scanner, I'm sure you'll see this doesn't recur. But again, this is just my own experience.
As for the second book, I'm not sure. I think whether what's happening there is the result of the way it was printed or the way it was scanned, it could potentially be improved using dewarp. However, note that it would be extremely laborious, since you'd have to adjust every page by hand, so consider whether it's worth your while. The automatic setting does exist, but it's by no means always right.
I did actually try it myself, and it seems that at least with that page, the results of the automatic dewarp are pretty good:
[HrXQpfe.zip](https://github.com/ScanTailor-Advanced/scantailor-advanced/files/11599031/HrXQpfe.zip)

But, again, look at that and decide for yourself if that's enough of an improvement to be worth the trouble. Having done this with entire books in the past, I can tell you it gets old fast.

Also, this concerns only ScanTailor Advance's dewarp function. I'm not sure, but I think more sophisticated modern solutions do exist, you'd just have to look for them.

**Comment by mrobe** [comment]
Thanks much for your guidance on this. 

I agree that doing it manually for many pages in a book is probably too much work (which is one reason I'm trying scan tailor in the first place!), and the automatic dewarp you show looks like a definite improvement.

So, I'll give it a go with automatic dewarp.

I'm still struggling a bit with the various other options (e.g., content box is cropping the page too small in some cases), but I guess this is just the learning curve.

**Comment by darnn** [comment]
Oh, yeah, you will have to adjust the content box yourself sometimes... Note that STA has a pretty handy (though not always perfect) feature where you can double-click the edge of the image you want the content box to jump to, and it will do it on its own, without you having to drag the edge to resize it yourself:
![sta-resize](https://github.com/ScanTailor-Advanced/scantailor-advanced/assets/19672444/c7546074-4159-41dd-aa79-d8e698dddb2b)

**Reply by mrobe** [reply]
Hmm. I am unable to get that to work.

For some reason, the original image is being cropped by both the page and the content box, and even if I manage to adjust the size, it doesn't show a larger image. For example: https://freeimage.host/i/HrhDPvs

None of the 'Page Box' or 'Content Box' options seem to work as expected. I.e., Page Box 'Auto' has no effect. and in 'Manual' mode I am unable to change the size of the page box.

What is the correct procedure? Is there some key I should press to get this to work, or... ?

Also, I don't understand why the page box is being manipulated at all. I want the same size pages for output of every page.

**EDIT: Please ignore. I finally figured out that 'Split Page' automatically decided that the page region was smaller. I disabled this and now I get the full page in 'Select Content'.**

**Comment by zvezdochiot** [comment]
@mrobe say:
>I agree that doing it manually for many pages in a book is probably too much work (which is one reason I'm trying scan tailor in the first place!), and the automatic dewarp you show looks like a definite improvement.

See also https://github.com/ImageProcessing-ElectronicPublications/photoquick filter "Deformation/Geometry/Un-tilt" witch option "Isometric".

And see https://github.com/ImageProcessing-ElectronicPublications/yasw
